test/test_engine.rb

```diff
@@ -54,15 +54,16 @@ def test_run
       engine=nil
       engine=Rutema::Engine.new(conf)
       engine.run
-      assert_equal(4, MockReporter.updates)
+      assert_equal(8, MockReporter.updates)
 
 
       conf[:tests]=["#{File.expand_path(File.dirname(__FILE__))}/data/sample.spec","#{File.expand_path(File.dirname(__FILE__))}/data/duplicate_name.spec"]
       assert_raise(Rutema::ParserError){
           engine=Rutema::Engine.new(conf)
           engine.run
       }
-      assert_equal(0, MockReporter.updates)
+      sleep(0.1)
+      assert_equal(2, MockReporter.updates)
```

Comment:
This test became wonky through the timing changes of the dispatching. The additional tiny delay ensures that the _start_ message and the _error_ message are both received by the mocked reporter.